server/datastore/datastore_app_test.go

```diff
@@ -75,3 +75,80 @@ func testAdditionalQueries(t *testing.T, ds kolide.Datastore) {
 	assert.Nil(t, err)
 	assert.JSONEq(t, `{"foo":"bar"}`, string(*info.AdditionalQueries))
 }
+
+func testEnrollSecrets(t *testing.T, ds kolide.Datastore) {
+	name, err := ds.VerifyEnrollSecret("missing")
+	assert.NotNil(t, err)
+	assert.Empty(t, name)
+
+	err = ds.ApplyEnrollSecretSpec(
+		&kolide.EnrollSecretSpec{
+			Secrets: []kolide.EnrollSecret{
+				kolide.EnrollSecret{Name: "one", Secret: "one_secret", Active: true},
+				kolide.EnrollSecret{Name: "two", Secret: "two_secret", Active: false},
+			},
+		},
+	)
+	assert.Nil(t, err)
+
+	name, err = ds.VerifyEnrollSecret("one") // secret name not actual secret
+	assert.NotNil(t, err)
+	assert.Empty(t, name)
+	name, err = ds.VerifyEnrollSecret("one_secret")
+	assert.Nil(t, err)
+	assert.Equal(t, "one", name)
+	name, err = ds.VerifyEnrollSecret("two_secret")
+	assert.NotNil(t, err)
+	assert.Equal(t, "", name)
+
+	err = ds.ApplyEnrollSecretSpec(
+		&kolide.EnrollSecretSpec{
+			Secrets: []kolide.EnrollSecret{
+				kolide.EnrollSecret{Name: "one", Secret: "one_secret", Active: false},
+				kolide.EnrollSecret{Name: "two", Secret: "two_secret", Active: true},
+			},
+		},
+	)
+	assert.Nil(t, err)
+
+	name, err = ds.VerifyEnrollSecret("one_secret")
+	assert.NotNil(t, err)
+	assert.Equal(t, "", name)
+	name, err = ds.VerifyEnrollSecret("two_secret")
+	assert.Nil(t, err)
+	assert.Equal(t, "two", name)
+
+}
+
+func testEnrollSecretRoundtrip(t *testing.T, ds kolide.Datastore) {
+	spec, err := ds.GetEnrollSecretSpec()
+	require.Nil(t, err)
+	assert.Len(t, spec.Secrets, 1)
+
+	expectedSpec := kolide.EnrollSecretSpec{
+		Secrets: []kolide.EnrollSecret{
+			kolide.EnrollSecret{Name: "one", Secret: "one_secret", Active: false},
+			kolide.EnrollSecret{Name: "two", Secret: "two_secret", Active: true},
+		},
+	}
+	err = ds.ApplyEnrollSecretSpec(&expectedSpec)
+	require.Nil(t, err)
+
+	spec, err = ds.GetEnrollSecretSpec()
+	require.Nil(t, err)
+	assert.Len(t, spec.Secrets, 3)
+	for _, secret := range spec.Secrets {
```

Comment:
Do you know a better pattern for this? I find myself doing this often and wishing there was a better way.

Reply:
You're trying to match `[]spec.Secrets` exactly, right?

Does `ElementsMatch` get you closer? Though last time I needed that, I ended up sorting and using `EqualValues` (see [launcher](https://github.com/kolide/launcher/blob/6ff84fba146ed3d2070faa30bd4947b2e16d7072/pkg/dataflatten/flatten_test.go#L349-L356))

If this can be represented in json, maybe convert to json, and compare that?

Reply:
Thank you for those ideas. The tricky thing is that the `created_at` date is set in the database so is not static with each test. For now I've gone with sorting and asserting on the known order.